com-api/src/sample_impl.rs

```diff
@@ -0,0 +1,211 @@
+use std::ops::{Deref, DerefMut};
+use std::path::Path;
+use std::time::Duration;
+
+pub struct RuntimeImpl {}
+impl crate::Runtime for RuntimeImpl {}
+
+pub struct RuntimeBuilderImpl {}
+
+impl crate::Builder for RuntimeBuilderImpl {
+    type Output = RuntimeImpl;
+    fn build(self) -> Self::Output {
+        Self::Output {}
+    }
+}
+
+impl crate::RuntimeBuilder for RuntimeBuilderImpl {
+    fn load_config(&mut self, _config: &Path) -> &mut Self {
+        self
+    }
+}
+
+impl RuntimeBuilderImpl {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+struct LolaEvent<T> {
+    event: PhantomData<T>,
+}
+
+struct LolaBinding<'a, T>
+where
+    T: Send,
+{
+    data: *mut T,
+    event: &'a LolaEvent<T>,
+}
+
+unsafe impl<'a, T> Send for LolaBinding<'a, T> where T: Send {}
+
+enum SampleBinding<'a, T>
+where
+    T: Send,
+{
+    Lola(LolaBinding<'a, T>),
+    Test(Box<T>),
+}
+
+pub struct Sample<'a, T>
+where
+    T: Reloc + Send,
+{
+    inner: SampleBinding<'a, T>,
+}
+
+impl<'a, T> From<T> for Sample<'a, T>
+where
+    T: Reloc + Send,
+{
+    fn from(value: T) -> Self {
+        Self {
+            inner: SampleBinding::Test(Box::new(value)),
+        }
+    }
+}
+
+impl<'a, T> Deref for Sample<'a, T>
+where
+    T: Reloc + Send,
+{
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        match &self.inner {
+            SampleBinding::Lola(_lola) => unimplemented!(),
+            SampleBinding::Test(test) => test.as_ref(),
+        }
+    }
+}
+
+impl<'a, T> super::Sample<T> for Sample<'a, T> where T: Send + crate::Reloc {}
+
+pub struct SampleMut<'a, T>
+where
+    T: Reloc,
+{
+    data: T,
+    _lifetime: PhantomData<&'a T>,
+}
+
+impl<'a, T> crate::SampleMut<T> for SampleMut<'a, T>
+where
+    T: Reloc + Send,
+{
+    type Sample = Sample<'a, T>;
+
+    fn into_sample(self) -> Self::Sample {
+        todo!()
+    }
+
+    fn send(self) -> crate::Result<()> {
+        todo!()
+    }
+}
+
+impl<'a, T> Deref for SampleMut<'a, T>
+where
+    T: Reloc,
+{
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.data
+    }
+}
+
+impl<'a, T> DerefMut for SampleMut<'a, T>
+where
+    T: Reloc,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.data
+    }
+}
+
+pub struct SampleMaybeUninit<'a, T>
+where
+    T: crate::Reloc + Send,
+{
+    data: MaybeUninit<T>,
+    _lifetime: PhantomData<&'a T>,
+}
+
+impl<'a, T> crate::SampleMaybeUninit<T> for SampleMaybeUninit<'a, T>
+where
+    T: Reloc + Send,
+{
+    type SampleMut = SampleMut<'a, T>;
+
+    fn write(self, val: T) -> SampleMut<'a, T> {
+        SampleMut {
+            data: val,
+            _lifetime: PhantomData,
+        }
+    }
+
+    fn as_mut_ptr(&mut self) -> *mut T {
+        todo!()
+    }
+
+    unsafe fn assume_init(self) -> SampleMut<'a, T> {
+        unsafe {
+            SampleMut {
+                data: self.data.assume_init(),
+                _lifetime: PhantomData,
+            }
+        }
```

Comment:
(misc) I would prefer to limit the scope of the `unsafe` to the least amount required:

```suggestion
        SampleMut {
            data: unsafe { self.data.assume_init() },
            _lifetime: PhantomData,
        }
```